addons/account/models/account_move.py

```diff
@@ -4206,8 +4215,9 @@ def _unlink_or_reverse(self):
             return
         to_reverse = self.env['account.move']
         to_unlink = self.env['account.move']
-        lock_date = self.company_id._get_user_fiscal_lock_date()
         for move in self:
+            # TODO: what about tax lock date?
```

Comment:
What if we only have the tax lock date but not the fiscalyear (or hard or sale or purchase) violated (and the lines have taxes). Should we not put it in `to_reverse` instead of `to_unlink`? (The unlinking of the lines will fail anyway I guess but still?)